src/mechanics/IGA/NURBS.h

```diff
@@ -0,0 +1,478 @@
+#pragma once
+
+#include<eigen3/Eigen/Dense>
+#include <array>
+#include <vector>
+#include <iostream>
+#include "mechanics/nodes/NodeSimple.h"
+
+namespace NuTo
+{
+template <int TDimParameter>
+class NURBS
+{
+public:
+
+    enum mParametrization {chord, centripetal};
+
+    /** Constructors **/
+
+    //! @brief ... default constructor
+    NURBS(){}
+
+    //! @brief ... constructor
+    //! @param rDegree ... degree of the polynomial
+    //! @param rKnots ... knot vector
+    //! @param rControlPoints ... control points
+    NURBS(const std::array<std::vector<double>, TDimParameter> &rKnots,
+          const std::vector<std::vector<NodeSimple>>           &rControlPoints,
+          const std::vector<std::vector<double>>               &rWeights,
+          const std::array<int,TDimParameter>                  &rDegree)
+        : mKnots(rKnots), mControlPoints(rControlPoints), mWeights(rWeights), mDegree(rDegree)
+    {}
+
+    /** Getter **/
+
+    //! @brief ... get the dimension of the curve
+    //! @return ... dimension of the curve
+    int GetDimension() const
```

Comment:
Maybe you could add a comment about what dimension `TDimParameter` is, and the dimension returned here.